remoteconfig/client.go

```diff
@@ -0,0 +1,223 @@
+
+	// Optional. Version number of the RemoteConfig to look up.
+	// If not specified, the latest RemoteConfig will be returned.
+	if versionNumber != "" {
+		opts = append(opts, internal.WithQueryParam("versionNumber", versionNumber))
+	}
+
+	var data RemoteConfig
+
+	resp, err := c.hc.DoAndUnmarshal(
+		context.Background(),
+		&internal.Request{
+			Method: http.MethodGet,
+			URL:    c.getRootURL(),
+			Opts:   opts,
+		},
+		&data,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &Response{
+		RemoteConfig: data,
+		Etag:         resp.Header.Get("Etag"),
+	}
+
+	return result, nil
+}
+
+// UpdateRemoteConfig https://firebase.google.com/docs/reference/remote-config/rest/v1/projects/updateRemoteConfig
+func (c *Client) UpdateRemoteConfig(eTag string, validateOnly bool) (*Response, error) {
+	if eTag == "" {
+		eTag = "*"
+	}
+
+	var opts []internal.HTTPOption
+	opts = append(opts, internal.WithHeader("If-Match", eTag))
+	opts = append(opts, internal.WithQueryParam("validateOnly", strconv.FormatBool(validateOnly)))
+
+	var data RemoteConfig
+
+	resp, err := c.hc.DoAndUnmarshal(
+		context.Background(),
+		&internal.Request{
+			Method: http.MethodPut,
+			URL:    c.getRootURL(),
+			Opts:   opts,
+		},
+		&data,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &Response{
+		RemoteConfig: data,
+		Etag:         resp.Header.Get("Etag"),
+	}
+
+	return result, nil
+}
+
+// ListVersions https://firebase.google.com/docs/reference/remote-config/rest/v1/projects.remoteConfig/listVersions
+func (c *Client) ListVersions(options *ListVersionsOptions) (*ListVersionsResponse, error) {
+	var opts []internal.HTTPOption
+	if options.PageSize != 0 {
+		opts = append(opts, internal.WithQueryParam("pageSize", strconv.FormatInt(options.PageSize, 10)))
+	}
+
+	if options.PageToken != "" {
+		opts = append(opts, internal.WithQueryParam("pageToken", options.PageToken))
+	}
+
+	if options.EndVersionNumber != "" {
+		opts = append(opts, internal.WithQueryParam("endVersionNumber", options.EndVersionNumber))
+	}
+
+	if options.StartTime != "" {
+		opts = append(opts, internal.WithQueryParam("startTime", options.StartTime))
+	}
+
+	if options.EndTime != "" {
+		opts = append(opts, internal.WithQueryParam("endTime", options.EndTime))
+	}
+
+	var data ListVersionsResponse
+	url := fmt.Sprintf("%s:listVersions", c.getRootURL())
+
+	_, err := c.hc.DoAndUnmarshal(
+		context.Background(),
+		&internal.Request{
+			Method: http.MethodGet,
+			URL:    url,
+			Opts:   opts,
+		},
+		&data,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &data, nil
+}
+
+// Rollback https://firebase.google.com/docs/reference/remote-config/rest/v1/projects.remoteConfig/rollback
+func (c *Client) Rollback(versionNumber string) (*Template, error) {
+	if versionNumber == "" {
+		return nil, errors.New("versionNumber is required to rollback a Remote Config template")
+	}
+
+	var data Template
+	url := fmt.Sprintf("%s:listVersions", c.getRootURL())
```

Comment:
I think this should be `rollback`.
```suggestion
	url := fmt.Sprintf("%s:rollback", c.getRootURL())
```